From 88a5a6f4bc75b89f1bb10af6ac0f318e529c5d98 Mon Sep 17 00:00:00 2001
From: Ariel Aviad <aaviad@habana.ai>
Date: Wed, 11 Jun 2025 12:44:20 +0300
Subject: [PATCH] [SW-231988] g2: device id: fix pcie device id

In gaudi2 HL288 card device id is gaudi2
device id but HL288 has its own device id.

This patch is setting the correct device id in
HL288 device

Change-Id: I5933dc27845ede863a874ac297e0d318f4c1400f
Signed-off-by: Ariel Aviad <aaviad@habana.ai>
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index a7350de..6525584 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -339,7 +339,12 @@
 void preboot_device_id_init(void)
 {
 	/* using vendor id as a magic number */
-	pci_dev_id = PCI_IDS_GAUDI2 | (PCI_VENDOR_ID_HABANALABS << 16);
+	if (hl_is_card_type_pcie())
+		pci_dev_id = PCI_IDS_GAUDI2_HL_288;
+	else
+		pci_dev_id = PCI_IDS_GAUDI2;
+
+	pci_dev_id |= (PCI_VENDOR_ID_HABANALABS << 16);
 }
 
 u32 preboot_device_id_get(void)
